### ISSUE TITLE
[Question] how can i get time at which i assign  some device to customer

### ISSUE THREAD

**Issue by securitylamb** [thingsboard/thingsboard#8491]
Component
 * Rule Engine

Description
I am trying to get time at which i assign device to the customer , as I see there is actually a event generated when we assign  device to customer in the audit logs I  wanted that timestamp and send that timestamp to my telemetry 

![image](https://user-images.githubusercontent.com/34526029/236375813-35b26103-f241-49a1-8a4f-ccaad657b8ec.png)

I want this timestamp to be send inside the telemetry data if possible using rule chain

**Comment by devaskim** [comment]
Try to use REST API node and [audit API](https://demo.thingsboard.io/swagger-ui/#/audit-log-controller)

**Comment by securitylamb** [comment]
> Try to use REST API node and [audit API](https://demo.thingsboard.io/swagger-ui/#/audit-log-controller)

but how will i pass  ACCESS CODE AND device id  dynamically inside that if there are no other options

**Comment by devaskim** [comment]
1. Unfortunately, TB doesn't support API token, so I use separate admin account to call TB REST API from Rule Chain. [Here ](https://github.com/devaskim/awesome-thingsboard/tree/main/chain/tb_api_token)is how I login and get access token from within rule chain.
2. Once you assign device to a customer, rule chain will process [ENTITY_ASSIGNED event](https://thingsboard.io/docs/user-guide/rule-engine-2-0/overview/#predefined-message-types). I guess this event should include useful information.

**Comment by securitylamb** [comment]
thanks a lot i was able to follow your instruction smoothly I just got stuck after the first step how will I pass device id to the telemetry  because audit log needs device id  and I don't have device id here

**Comment by devaskim** [comment]
So did you got answer for your question? If yes, suggest to close the issue

**Comment by securitylamb** [comment]
as mentioned i am able to login using rule chain but my main question still goes unanswered  how will i dynamically pass device id when someone clicks on a particular device in the dashboard

**Comment by devaskim** [comment]
Sorry, but the main question is that which is in the title of the issue. The original post also doesn't include any information about `someone clicks on a particular device in the dashboard`. Thus either provide more information of your whole desired flow, or create another issue if I answered for question that is in the title.

**Comment by securitylamb** [comment]
again the issue is not with login it is with device logs and for the getting device logs it also need device id along with access token

**Comment by devaskim** [comment]
> I wanted that timestamp and send that timestamp to my telemetry

I guess `my` means the telemetry of device which was assigned to customer. Once you or someone else assign(s) a device to a customer, rule chain should push telemetry packet to this device with time stamp of assignment. Right?

**Comment by securitylamb** [comment]
yes

**Comment by securitylamb** [comment]
yes i want the timestamp at which it was assigned as shown in the above picture

**Comment by devaskim** [comment]
Did you try my solution?
